Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -507,6 +529,162 @@ private byte[] GetKey(ShortHash hash)
             return hash.ToByteArray();
         }
 
+        #region Memoization
+        // TODO(jubayard): garbage collection / removal in general
+
+        /// <inheritdoc />
+        public override GetContentHashListResult GetContentHashList(OperationContext context, StrongFingerprint strongFingerprint)
+        {
+            var key = GetCaChaaSKey(strongFingerprint);
+            ContentHashListWithDeterminism? result = null;
+            var status = _keyValueStore.Use(
+                store =>
+                {
+                    if (store.TryGetValue(key, out var data, nameof(Columns.CaChaaS)))
+                    {
+                        result = DeserializeContentHashList(data);
+                        // TODO(jubayard): since we are inside the ContentLocationDatabase, we can validate that all
+                        // hashes exist. Moreover, we can prune content.
+                    }
+                });
+
+            if (!status.Succeeded)
+            {
+                return new GetContentHashListResult(status.Failure.CreateException());
+            }
+            
+            return new GetContentHashListResult(result ?? default);
+        }
+        
+        /// <inheritdoc />
+        public override AddOrGetContentHashListResult AddOrGetContentHashList(OperationContext context, StrongFingerprint strongFingerprint, ContentHashListWithDeterminism contentHashListWithDeterminism)
+        {
+            var key = GetCaChaaSKey(strongFingerprint);
+
+            Possible<AddOrGetContentHashListResult> status = _keyValueStore.Use(

```

Comment:
Let's be consistent (or at least let's try) in terms of var vs explicit types.

I prefer 'var' unless you really want to be explicit, but I see that in this method you have explicit type but in GetContentHashList var is used.

Reply:
I am being consistent. All other places use Possible<Unit>; this case is specified only because the type is different than what you would usually expect this to return.

---
In reply to: [302175159](https://github.com/microsoft/BuildXL/pull/571#discussion_r302175159) [](ancestors = 302175159)

Reply:
The question is: do you need to know this type? I think I'm fine with using var here.

---
In reply to: [302228489](https://github.com/microsoft/BuildXL/pull/571#discussion_r302228489) [](ancestors = 302228489,302175159)